utils/time.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package utils
+
+import "time"
+
+func TruncateToDay(t time.Time) time.Time {
+	return time.Date(
+		t.Year(),
+		t.Month(),
+		t.Day(),
+		0,
+		0,
+		0,
+		0,
+		t.Location(),
+	)
+}
```

Comment:
TIP: `time` already has this in the standard library
```go
date := t.Truncate(24 * time.Hour)
// Or, to use UTC as our reference TZ
date = t.UTC().Truncate(24 * time.Hour)
```

Reply:
we had this discussion :) truncate does truncate, not resetting the date fields. besides we already use it in other services :)